mp/src/game/shared/neo/neo_gamerules.cpp

```diff
@@ -1962,6 +1960,22 @@ void CNEORules::ClientDisconnected(edict_t* pClient)
 			ghost->SetRemoveable(false);
 			pNeoPlayer->Weapon_Detach(ghost);
 		}
+
+		// Save XP/death counts
+		if (neo_sv_player_restore.GetBool())
+		{
+			const CSteamID steamID = GetSteamIDForPlayerIndex(pNeoPlayer->entindex());
+			if (steamID.IsValid())
+			{
+				const uint32 accountID = steamID.GetAccountID();
+				const RestoreInfo restoreInfo{
+					.xp = pNeoPlayer->m_iXP.Get(),
+					.deaths = pNeoPlayer->DeathCount(),
```

Comment:
Nit: call `Remove` if xp == 0 and deaths == 0 and skip `Insert`, because that's the default state of things and there's no need to cache it.